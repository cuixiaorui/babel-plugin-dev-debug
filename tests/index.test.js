const babel = require("@babel/core");
const plugin = require("../");
const inputCode = `
        const a = 10;
        const b = 20;

        if(DEBUG){
            console.log("heihei")
        }
        `;

describe("babel-plugin-dev-debug", () => {
  it("dev", () => {
    // input
    process.env.NODE_ENV = "development";

    const { code } = babel.transform(inputCode, { plugins: [plugin] });
    expect(code).toMatchSnapshot();
  });

  it("prod", () => {
    // input
    process.env.NODE_ENV = "production";

    const { code } = babel.transform(inputCode, { plugins: [plugin] });
    expect(code).toMatchSnapshot();
  });
});
