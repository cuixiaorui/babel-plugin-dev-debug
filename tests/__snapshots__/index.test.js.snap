// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`babel-plugin-dev-debug dev 1`] = `
"const a = 10;
const b = 20;

if (\\"DEBUG\\") {
  console.log(\\"heihei\\");
}"
`;

exports[`babel-plugin-dev-debug prod 1`] = `
"const a = 10;
const b = 20;"
`;
